### DISCUSSION TITLE
Cingular pole from grey matter meshes?

### DISCUSSION THREAD

**Discussion by AnnaBhlr** [brainvisa/brainvisa.github.io#157] [Q&A]
Hi and thanks for helping me so far,

I was wondering if there is a way to get the vertices belonging to the cingular pole in grey matter meshes (which are a result from the Morphologist pipeline)? Similar to the labeling of Freesurfer meshes...

Again, thanks a lot!

**Comment by denisri** [comment]
Hi,
For that you can use the "Cingular pole projection" process (which is part of the HipHop pipeline), in cortical_surface/anatomy/tools/ in BrainVisa. The cingular pole labeling is perhaps not perfect because its aim there is just to find the center of the pole in order to determine a coordinates system origin on the brain mesh, not to label exactly the cingular pole boundaries.
Denis

**Reply by AnnaBhlr** [reply]
Thanks! I saw that CingularPoleProjection needs a lot of inputs. As I did not preprocess the T1w scan myself I'm not sure where to find the subject_transformation, pole_template and themplate_pole_transformation needed, also not sure what the needed integers should be...
I also saw the process CingularPoleFromFreesurfer which handles less input. Would that also be an option to get the cingular pole labeled in the meshes? If, what is needed as Freesurfer directory?

**Comment by denisri** [comment]
This process is meant to be run after the Morphologist pipeline. If the T1w has been processed using Morphologist, then selecting the white mesh as the first input should be enough to complete all other parameters automatically. The template parameters should take data from the BrainVisa shared data (not subject-specific).

**Reply by AnnaBhlr** [reply]
I think we narrowing down my problem. CingularPoleProjection is not running with just the white mesh as input. This is the error I got:

KeyError: 'Could not get process "\'SingularPoleProjection\'": invalid identifier or process file: IOError(2, \'No such file or directory\')'

I just the same Docker container which was used for processing the data (this one: https://github.com/LeonieBorne/morpho-deepsulci-docker). I reckon the BrainVisa shared data is not in there as didn't found I folder named 'SingularPoleProjection' in the Docker container?

I really would like to build up on the work Leonie has done!

**Comment by denisri** [comment]
CingularPoleProjection is written with a "C", not "S", would it just be the solution ? ;)
Otherwise I have not checked Leonie's docker container, so I don't know if all toolboxes are included there. However this container is 3 or4 years old, and was released (as far as I remember) because BrainVisa standard distribution was not released yet by that time with her sulci identification method. This has been done now, so maybe the standard BrainVisa 5.1 containers would do the job for you ?

**Reply by AnnaBhlr** [reply]
oh my gosh! It was just the spelling mistake! Thank you so much for your patience.

**Comment by gauzias** [comment]
Hi @AnnaBhlr 
As pointed by @denisri this tool needs several files as input that correspond to classical outputs from the morphologist pipeline.
If your mesh was not produced using morphologist, then it will be quite challenging to get good results.
More specifically, you need the appropriate affine transformations that will serve to put the individual mesh in the referential of the template.
And these transformations are automatically computed during the morphologist pipeline.

**Reply by AnnaBhlr** [reply]
Hi @gauzias,
Thanks for your answer!  My meshes are a product from the Morphologist pipeline and I do have the subject specific transformation files. 

That's how I ran the CingularPoleProjection:

`axon-runprocess CingularPoleProjection white_mesh=/path/tosubject/t1mri/default_acquisition/default_analysis/segmentation/mesh/subjectnumber_Lwhite.gii side='left' subject_transformation=/path/tosubject/t1mri/default_acquisition/registration/RawT1-subjectnumber_default_acquisition_TO_Talairach-ACPC.trm pole_template=/home/brainvisa/build/bug_fix/share/brainvisa-share-4.6/hemitemplate/Template_Cingular.nii.gz template_pole_transformation=/home/brainvisa/build/bug_fix/share/brainvisa-share-4.6/hemitemplate/RawT1-Template_TO_Talairach-ACPC.trm pole=/data/testpoletexture.gii"`

And that's the results (blue dots correspond to the results from the process). 

<img width="368" alt="Bildschirm­foto 2023-11-17 um 11 30 55" src="https://github.com/brainvisa/brainvisa.github.io/assets/131627125/cd46ba1c-65e5-4b98-ade1-bc80a1669830">

**Comment by gauzias** [comment]
Hi @AnnaBhlr 
the most probable cause would be that the transformation 'subject_transformation' that was computed by the morphologist pipeline is not sufficiently accurate.
When it works well, you get something similar to the image attached.
So I suggest to check the computation of the transformation in the morphologist pipeline.

![Screenshot from 2023-11-22 16-13-49](https://github.com/brainvisa/brainvisa.github.io/assets/43206340/1344abee-4654-48c3-a691-f793d3cb3296)

**Reply by AnnaBhlr** [reply]
Hi @gauzias,

Thanks for your answer! The plot you posted is exactly what I'm aiming for. 

I had a look at the subject transformation file but it looks alright to me. Any ideas for debugging? The problem is that I'm sitting on approx. 3000 subject meshes from the morphologist pipeline and I'd like to use them for further analysis. It would be excellent to find a solution within Brainvisa especially as providing labeled meshes is done by other pipeline. I know that I could re-run the T1w images in Freesurfer to get labeled meshes but I'd lost some benefits of the BrainVisa pipeline.
